internal/testutil/retry.go

```diff
@@ -0,0 +1,81 @@
+// Copyright 2016 Google Inc. All rights reserved.
+// Use of this source code is governed by the Apache 2.0
+// license that can be found in the LICENSE file.
+
+package testutil
+
+import (
+	"bytes"
+	"fmt"
+	"path/filepath"
+	"runtime"
+	"strconv"
+	"testing"
+	"time"
+)
+
+// Flaky runs function f for up to maxAttempts times until f returns successfully, and reports whether f was run successfully.
+// It will sleep for the given period between invocations of f.
+// Use the provided *testutil.R instead of a *testing.T from the function.
+func Flaky(t *testing.T, maxAttempts int, sleep time.Duration, f func(r *R)) bool {
```

Comment:
Do we need to retry anything other than flaky tests? If not, I'd rather call it Flaky. It makes the intent clearer.